### ISSUE TITLE
【FR】Bug: operator accuracy degradation on nvidia (Qwen/Qwen2.5-1.5B-Instruct)

### ISSUE THREAD

**Issue by Lxiparer** [FlagOpen/FlagGems#4553]
### Description

Accuracy evaluation on Qwen2.5-1.5B-Instruct shows **10% absolute accuracy drop** (V1 native=32%, V2 FlagGems=22%, threshold=5.0%). Performance is healthy (95.2% of native).

Accuracy tuning ran 3 rounds of operator group disabling, but all rounds scored **18% — worse than the full FlagGems baseline of 22%**. The 12 tested operators are ruled out; the root cause lies in the remaining 23 operators that were never disabled.

### Environment

| Item | Value |
|------|-------|
| Hardware | nvidia |
| GPU | NVIDIA H20-3e |
| PyTorch | 2.11.0+cu130 |
| FlagGems | 5.0.1.rc.0 |
| vLLM | 0.20.2 |

### Operator Diagnosis Results

Accuracy tuning (`traces/05_accuracy_tuning.json`, `logs/pipeline.log`):

| Round | Group | Disabled Operators | Score | Drop | Result |
|-------|-------|--------------------|-------|------|--------|
| 1 | compute | `addmm` | 18.0% | 14.0% | FAILED |
| 2 | compute+memory | `addmm, zeros, ones, ones_like, full` | 18.0% | 14.0% | FAILED |
| 3 | all 3 groups | `add, addmm, cos, div, full, mul, ones, ones_like, reciprocal, sin, sub, zeros` | 18.0% | 14.0% | FAILED |

**Key finding**: Disabling operators consistently produced **worse** accuracy (18%) than full FlagGems (22%). This means the accuracy issue is not caused by a single bad operator producing wrong outputs — it is caused by an **interaction effect**: certain FlagGems operators produce results that downstream native operators handle differently.

**Ruled-out operators** (12, disabling them did not help): `add, addmm, cos, div, full, mul, ones, ones_like, reciprocal, sin, sub, zeros`

**Remaining suspect operators** (23, never disabled): `arange, argmax, cat, copy, eq, exponential_, fill, gt, index, lt, masked_fill, pow, rand_like, scatter, scatter_add_, softmax, to, where, general_mm, randn, sort, cumsum, gather, le`

Of these, the most likely candidates for numerical interaction issues:
- `softmax` (`flag_gems.ops.softmax.softmax`) — FP32/FP16 accumulation differences
- `general_mm` (`flag_gems.runtime.backend._nvidia.hopper.ops.mm.general_mm`, shape [-, 8192, 1536, 1536]) — large matmul precision
- `masked_fill` / `scatter_add_` — conditional operations sensitive to FP16 rounding

### Steps to Reproduce

1. Environment: nvidia H20-3e, PyTorch 2.11.0+cu130, FlagGems 5.0.1.rc.0, vLLM 0.20.2
2. Enable FlagGems: `flag_gems.enable(record=True, once=True, path="/root/gems.txt")`
3. Run GPQA evaluation (thinking mode) with 35 operators active
4. Observe V1=32% → V2=22% (10% drop)
5. Run `diagnose_ops.py accuracy-groups` to isolate problematic operator groups
6. Note that disabling up to 12 operators made accuracy worse, not better

### Actual Behavior

```
V1 (native PyTorch)   : 32.0%
V2 (full FlagGems)    : 22.0%  ← 10% drop
V2 (disable addmm)    : 18.0%  ← even worse
V2 (disable 5 ops)    : 18.0%
V2 (disable 12 ops)   : 18.0%
```

### FlagGems Integration Code

**File**: `/usr/local/lib/python3.12/dist-packages/vllm/v1/worker/gpu_model_runner.py`
**Enable call**: `flag_gems.enable(record=True, once=True, path="/root/gems.txt")`

**Active operators** (35 in gems.txt): zeros, full, zero_, arange, div, pow, reciprocal, mul, cos, sin, cat, to_copy, ones, fill_scalar_, addmm_out, general_mm (hopper, [-,8192,1536,1536]), index, rand_like, argmax, lt_scalar, where_self, where_self_out, true_divide_, softmax, exponential_, copy_, add, sub, eq_scalar, masked_fill_, ones_like, scatter_add_0, gt_scalar, sub_, scatter_

### Possible Directions

- Investigate **operator interaction effects** — individual operators may be correct in isolation but produce accumulated errors when mixed with native fallbacks
- Prioritize: `softmax` (FP16 accumulation), `general_mm` (large matmul precision), `masked_fill`/`scatter_add_` (conditional FP16 ops)
- Run pair-disabling: keep one suspect operator enabled at a time with all others native, to isolate which single operator causes the 22%→18% degradation when combined with native fallbacks
- Check whether the 18% floor is a separate issue (e.g., GPQA evaluation artifact with specific operator combinations)
